### PR TITLE
tiltfile2: resolve docker-compose path relative to tiltfile root

### PR DESCRIPTION
**Pull request by jazzdan**
Also introduce the concept of a tiltfileRoot which could, in the future,
allow the Tiltfile and the directory we operate in to be disjoint paths.

### CODE REVIEW

File: internal/tiltfile2/tiltfile_state.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type tiltfileState struct {
 	// set at creation
 	ctx      context.Context
 	filename string
+	root     string
```
Comment (nicks) [inline internal/tiltfile2/tiltfile_state.go]: hmmm...doesn't TiltfileState already have a function for this? it should be absPath or smth?

Comment (jazzdan) [reply internal/tiltfile2/tiltfile_state.go]: We have functions for dealing with the paths for static (image) builds, but not one for the Tiltfile afaict. Got a link?

Comment (jazzdan) [reply internal/tiltfile2/tiltfile_state.go]: Aha! it was in the `files` file.

File: internal/tiltfile2/docker_compose.go [hunk 2]
```diff
@@ -24,7 +25,10 @@ func (s *tiltfileState) dockerCompose(thread *skylark.Thread, fn *skylark.Builti
 	if err != nil {
 		return nil, err
 	}
-	absConfigPath, err := filepath.Abs(configPath)
+	if !filepath.IsAbs(configPath) {
```
Comment (nicks) [inline internal/tiltfile2/docker_compose.go]: this is equivalent to configPath = s.absPath(configDir)

**Pull request merged**